presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -753,6 +766,51 @@ private static void buildColumnInfo(ConnectorTableMetadata tableMetadata, Immuta
         }
     }
 
+    private static Type toOutputType(TypeManager typeManager, Type type)
+    {
+        if (type.equals(BOOLEAN)) {
+            return type;
+        }
+        if (type.equals(BIGINT)) {
+            return type;
+        }
+        if (type.equals(DOUBLE)) {
+            return type;
+        }
+        if (type instanceof VarcharType) {
+            return VARCHAR;
+        }
+        if (type.equals(VARBINARY)) {
+            return type;
+        }
+        if (type.equals(DATE)) {
+            return type;
+        }
+        if (type.equals(TIMESTAMP)) {
+            return type;
+        }
+        if (isArrayType(type)) {
```

Comment:
It might be confusing which comparison to use for which type. Shouldn't there be one convention for this? Like each type should provide "static boolean isTypeName(Type type)" method?